spicy/src/rt/tests/zlib.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+
+#include <doctest/doctest.h>
+
+#include <hilti/rt/extension-points.h>
+#include <hilti/rt/types/bytes.h>
+
+#include <spicy/rt/zlib.h>
+
+using namespace hilti::rt;
+using namespace hilti::rt::bytes::literals;
+using namespace spicy::rt;
+
+TEST_SUITE_BEGIN("Zlib");
+
+TEST_CASE("decompress") {
+    zlib::Stream stream;
+
+    SUBCASE("Bytes") {
+        SUBCASE("nothing") {
+            CHECK_EQ(zlib::decompress(stream, ""_b), ""_b);
+            CHECK_EQ(zlib::finish(stream), ""_b);
+        }
+
+        SUBCASE("single block") {
+            CHECK_EQ(zlib::decompress(stream, "x\x01\x01\x03\x00\xfc\xff\x00\x01\x02\x00\x07\x00\x04"_b),
+                     "\x00\x01\x02"_b);
+            CHECK_EQ(zlib::finish(stream), ""_b);
+        }
+
+        SUBCASE("multiple blocks") {
+            CHECK_EQ(zlib::decompress(stream, "x\x01\x01\x03\x00\xfc\xff\x00\x01\x02\x00\x07\x00\x04"_b),
```

Comment:
This is the same as in the "single block" case?

Reply:
Ups it is. I already broke the data up, but forgot to decompress it in two steps. Fixed now.